pennylane/transforms/adjoint.py

```diff
@@ -113,11 +113,17 @@ def circuit():
 
     @wraps(fn)
     def wrapper(*args, **kwargs):
-        with get_active_tape().stop_recording(), QuantumTape() as tape:
-            fn(*args, **kwargs)
-        for op in reversed(tape.queue):
+
+        if isinstance(fn, QuantumTape):
+            tape = fn(*args, **kwargs).copy()

```

Comment:
@rmoyard could it be that this line is not covered by the tests?

Reply:
@antalszava Yes I still have one problem with the new solution, hopefully it will be ok later today!